### ISSUE TITLE
RFC: Have yapf manage '\' line breaks

### ISSUE THREAD

**Issue by zaquestion** [google/yapf#344]
I understand that modifiying the token stream is generally discourage and otherwise nasty buisness in this project.
ex: **"Please manage commas"**: https://github.com/google/yapf/issues/274

There was talk in the referenced (similar) ticket that changes would be considered provided they were behind a knob and thoroughly tested. Lately much of my time has been in a code base with somewhat "eagar" line breaking. Many of which use `\` for such.
```
foo = \
        'bar'
vs
foo = 'bar'
```
yapf has done an amazing job at combining these unneeded breaks, but currently doesn't get mess with the above case. 

My proposal (if any) would be to add support to remove `\` where the lines would not break the defined column length. yapf seems to deal with the case where the lines are too long just fine so I don't see any reason to actually add them. With hope create an easy/safe way to handle this,#274 and related cases. Is this something of interest here, would a PR be consider given knob/s and tests.?

For my cases I can _probably_ just toss a `sed` in front of the yapf call, but it would be nice if the tool owned this logic. yapf has become a standard tooling for the python projects at our work and its fairly easy to sell adding/changing a knob.

**Comment by advance512** [comment]
Agreed. As far as I can see, the token stream isn't really modified - the \ symbols should not even be part of it. \ symbols simply mean - ignore the next newline character.

This seems related to #348.

**Comment by advance512** [comment]
You can verify that \'s indeed get removed (and thus can also be added) by formatting the following code:

```
rawKPIs = \
    KPIsSession.query(KPI.name, KPI.value, KPI.created) \
    .filter(KPI.entryId == entryID) \
    .filter(KPI.subsetId is None) \
    .filter(KPI.created >= fromTimestamp) \
    .filter(KPI.created <= toTimestamp) \
    .filter(KPI.created % moduloSelector == 0) \
    .order_by(KPI.name, KPI.created) \
    .all()
```

**Comment by gwelymernans** [comment]
YAPF needed to handle the `\` case in a unique way. Right now, it simply assumes that the programmer coded the line in the correct format and doesn't wish it to be touched. So YAPF doesn't touch lines with `\`s in them and emits them with the original formatting.

If you want to write a PR that can manage `\`s in a nicer way, I would be open to reviewing it. It *should* be easier to get "correct" than commas, because there's only one purpose for a `\` terminator on a line, and removing it doesn't affect the semantics of the program (of course assuming that the line is properly concatenated after removal).

If you want to get started on this, you should look at the `format_token` module. A `\` is a "continuation" token, and is used in special ways both during formatting and in the `unwrapped_line` module. It might be a simple matter of removing (or adding) that token to the `unwrapped_line` object. However, when you do this, the line spacing needs to be adjusted among the other `format_token` objects on the same line.

Please place the new feature behind a knob. (Actually, most current features need to be behind knobs... :-)
